java/core/src/test/java/co/worklytics/psoxy/rules/RulesBaseTestCase.java

```diff
@@ -233,13 +230,36 @@ protected void assertUrlBlocked(String url) {
 
     /**
      * Utility method to print out formatted JSON for debug easily
+     *
+     *
+     *
+     *
      * @param json
      * @return
      */
+    @SneakyThrows
     @SuppressWarnings("unused")
     protected String prettyPrintJson(String json) {
-        Gson gson = new GsonBuilder().setPrettyPrinting().create();
-        return gson.toJson(JsonParser.parseString(json));
+
+        return jsonMapper.writerWithDefaultPrettyPrinter().writeValueAsString(jsonMapper.readerFor(Object.class).readValue(json));
+
+        //NOTE: Gson seems to URL-encode embedded strings!?!?!
+        //  eg "64123avdfsMVA==" --> "64123avdfsMVA\u0030\0030"
```

Comment:
Probably needs to use the discardHTML flag -> https://www.javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/GsonBuilder.html#disableHtmlEscaping()

FYI https://stackoverflow.com/questions/11149931/gson-tojson-object-contain-url

Reply:
it's just weird; we should stick with Jackson, which is what we use everywhere else